nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmHostedRecipe.groovy

```diff
@@ -0,0 +1,259 @@
+import org.sonatype.nexus.repository.view.handlers.ContentHeadersHandler
+import org.sonatype.nexus.repository.view.handlers.ExceptionHandler
+import org.sonatype.nexus.repository.view.handlers.FormatHighAvailabilitySupportHandler
+import org.sonatype.nexus.repository.view.handlers.HandlerContributor
+import org.sonatype.nexus.repository.view.handlers.HighAvailabilitySupportChecker
+import org.sonatype.nexus.repository.view.handlers.TimingHandler
+import org.sonatype.nexus.repository.view.matchers.ActionMatcher
+import org.sonatype.nexus.repository.view.matchers.LiteralMatcher
+import org.sonatype.nexus.repository.view.matchers.logic.LogicMatchers
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher
+import org.sonatype.repository.helm.internal.AssetKind
+import org.sonatype.repository.helm.internal.HelmFormat
+import org.sonatype.repository.helm.internal.content.HelmContentFacet
+import org.sonatype.repository.helm.internal.content.createindex.CreateIndexFacetImpl
+import org.sonatype.repository.helm.internal.security.HelmSecurityFacet
+
+import static org.sonatype.nexus.repository.http.HttpMethods.DELETE
+import static org.sonatype.nexus.repository.http.HttpMethods.GET
+import static org.sonatype.nexus.repository.http.HttpMethods.HEAD
+import static org.sonatype.nexus.repository.http.HttpMethods.PUT
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_INDEX
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_PACKAGE
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_PROVENANCE
+
+/**
+ * Helm Hosted Recipe
+ *
+ * @since 1.0.11
+ */
+@Named(HelmHostedRecipe.NAME)
+@Singleton
+class HelmHostedRecipe
+    extends RecipeSupport
+{
+  public static final String NAME = 'helm-hosted'
+
+  @Inject
+  HostedHandlers hostedHandlers
+
+  @Inject
+  Provider<CreateIndexFacetImpl> createIndexFacet
+
+  @Inject
+  Provider<HelmContentFacet> helmFacet
+
+  @Inject
+  Provider<HelmHostedFacetImpl> hostedFacet
+
+  @Inject
+  Provider<HelmSecurityFacet> securityFacet
+
+  @Inject
+  Provider<ConfigurableViewFacet> viewFacet
+
+
+  @Inject
+  ExceptionHandler exceptionHandler
+
+  @Inject
+  TimingHandler timingHandler
+
+  @Inject
+  SecurityHandler securityHandler
+
+  @Inject
+  PartialFetchHandler partialFetchHandler
+
+  @Inject
+  ConditionalRequestHandler conditionalRequestHandler
+
+  @Inject
+  ContentHeadersHandler contentHeadersHandler
+
+  @Inject
+  HandlerContributor handlerContributor
+
+  @Inject
+  Provider<HttpClientFacet> httpClientFacet
+
+  @Inject
+  FormatHighAvailabilitySupportHandler formatHighAvailabilitySupportHandler
+
+  @Inject
+  HighAvailabilitySupportChecker highAvailabilitySupportChecker
+
+  @Inject
+  HelmHostedRecipe(@Named(HostedType.NAME) final Type type, @Named(HelmFormat.NAME) final Format format) {
+    super(type, format)
+  }
+
+  @Override
+  boolean isFeatureEnabled() {
+    return highAvailabilitySupportChecker.isSupported(getFormat().getValue())
```

Comment:
Do we really support high availability for the new DB? Maybe we should hardcode false here?

Reply:
Make sense, fixed 88bac16ecc93e6c7eb6f3fde288232ef8ea0468c

Reply:
Unfortunately ^ does not work. False means "format/recipe is unavailable".  isFeatureEnabled() should return true always. Raw was implemented in such way. So I assuming that highAvailability flags should not impact on repo functionality as far clustering is not supported for new DBs.